lib/tramway/base_form.rb

```diff
@@ -20,38 +22,13 @@ def initialize(object)
     class << self
       def inherited(subclass)
         subclass.instance_variable_set(:@properties, [])
+        subclass.instance_variable_set(:@normalizations, __ancestor_normalizations)
 
         super
       end
 
-      def property(attribute)
-        @properties << attribute
-
-        delegate attribute, to: :object
-      end
-
-      def properties(*attributes)
-        attributes.any? ? __set_properties(attributes) : __properties
-      end
-
-      def __set_properties(attributes)
-        attributes.each do |attribute|
-          property(attribute)
-        end
-      end
-
-      def __properties
-        (__ancestor_properties + @properties).uniq
-      end
-
-      # :reek:ManualDispatch { enabled: false }
-      def __ancestor_properties(klass = superclass)
-        superklass = klass.superclass
-
-        return [] unless superklass.respond_to?(:properties)
-
-        klass.properties + __ancestor_properties(superklass)
-      end
+      include Tramway::Forms::Properties
+      include Tramway::Forms::Normalizations
```

Comment:
this looks clean!
It could go even further: move parts inside `def inherited(subclass)`, and `__apply_normalizations` and `__apply_properties` also to these mixin classes

So they will contain whole isolated features

Reply:
Thanks :slightly_smiling_face: Done!